tools/tsp-client/ci.yml

```diff
@@ -0,0 +1,120 @@
+  paths:
+    include:
+      - tools/tsp-client
+
+pr:
+  branches:
+    include:
+      - main
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - tools/tsp-client
+
+variables:
+  - template: ../../eng/pipelines/templates/variables/globals.yml
+  - name: NodeVersion
+    value: '14.x'
+  - name: VAR_ARTIFACT_NAME
+    value: 'drop'
+  - name: VAR_BUILD_ARTIFACT_STAGING_DIRECTORY
+    value: $(Build.ArtifactStagingDirectory)
+
+stages:
+  - stage: InstallAndBuild
+    jobs:
+      - job: Build
+        strategy:
+          matrix:
+            linux:
+              imageName: ubuntu-22.04
+            mac:
+              imageName: 'macos-latest'
+            windows:
+              imageName: 'windows-latest'
+        pool:
+          vmImage: $(imageName)
+        steps:
+        - task: NodeTool@0
+          inputs:
+            versionSpec: '$(NodeVersion)'
+          displayName: 'Install Node.js'
+
+        - script: |
+            npm install
+          displayName: 'npm install'
+          workingDirectory: $(System.DefaultWorkingDirectory)/tools/tsp-client
+
+        - script: |
+            npm pack
+          displayName: 'npm pack'
+          workingDirectory: $(System.DefaultWorkingDirectory)/tools/tsp-client
+          condition: contains(variables['imageName'], 'ubuntu')
+
+        - script: 'cp azure-tools-tsp-client-*.tgz $(VAR_BUILD_ARTIFACT_STAGING_DIRECTORY)'
+          displayName: 'copy to staging dir'
+          workingDirectory: $(System.DefaultWorkingDirectory)/tools/tsp-client
+          condition: contains(variables['imageName'], 'ubuntu')
+
+        - task: PublishBuildArtifacts@1
+          inputs:
+            PathtoPublish: '$(VAR_BUILD_ARTIFACT_STAGING_DIRECTORY)'
+            ArtifactName: '$(VAR_ARTIFACT_NAME)'
+            publishLocation: 'Container'
+          condition: contains(variables['imageName'], 'ubuntu')
+
+  - ${{if ne(variables['Build.Reason'], 'PullRequest')}}:
+    - stage: Release
+      dependsOn: InstallAndBuild
+      condition: succeeded()
+      jobs:
+      - job: approve
+        pool: server
+        steps:
+        - task: ManualValidation@0
+          inputs:
+            notifyUsers: 'Click to approve if it''s an expected public release.'
+      - job: release
+        dependsOn: approve
+        condition: and(succeeded(), ne(variables['USER_SKIP_PUBLIC_RELEASE'], 'true'))
+        steps:
+        - task: NodeTool@0
+          inputs:
+            versionSpec: '$(NodeVersion)'
+          displayName: 'Install Node.js'
+        - task: DownloadBuildArtifacts@0
+          inputs:
+            buildType: 'current'
+            downloadType: 'single'
+            artifactName: '$(VAR_ARTIFACT_NAME)'
+            downloadPath: '$(VAR_BUILD_ARTIFACT_STAGING_DIRECTORY)'
+        - script: |
+            echo -e "\e[32m[$(date -u)] LOG: Ensure latest npm is installed"
+            sudo npm install -g npm || { echo 'npm install failed' ; exit 1; }
```

Comment:
Does `sudo` work on the pipeline agents? I would think we want to avoid any use of sudo here.

Reply:
I told @catalinaperalta to copy/paste this form one of the other pipelines. Eventually I want @ckairen to add a common publishing template we can use in the tools repo for all the npm package publishing.

Reply:
We should just use the default version of `npm` bundled with whatever version of `node` we are using.

I don't see any reason to install `pnpm`, since it's not used later in this script.